src/settings-ui/Settings.UI/Views/AwakePage.xaml

```diff
@@ -49,9 +49,20 @@
                             <ComboBoxItem x:Uid="Awake_NoKeepAwake" />
                             <ComboBoxItem x:Uid="Awake_IndefiniteKeepAwake" />
                             <ComboBoxItem x:Uid="Awake_TemporaryKeepAwake" />
+                            <ComboBoxItem x:Uid="Awake_ExpirableKeepAwake" />
                         </ComboBox>
                     </labs:SettingsCard>
 
+                    <labs:SettingsCard
+                        x:Uid="Awake_ExpireAtAwake"
+                        HeaderIcon="{ui:FontIcon FontFamily={StaticResource SymbolThemeFontFamily}, Glyph=&#xEC92;}"
+                        Visibility="{x:Bind ViewModel.IsExpirationConfigurationEnabled, Mode=OneWay}">
+                        <StackPanel Orientation="Horizontal">
+                            <DatePicker Date="{x:Bind ViewModel.ExpireAt, Mode=TwoWay}"></DatePicker>
+                            <TimePicker Margin="8,0,0,0" Time="{x:Bind ViewModel.Time, Mode=TwoWay}"></TimePicker>
```

Comment:
Additional suggestions:

- `Awake_ExpireAtAwake` or `Awake_ExpireAtDateTime`?
- why not name the two `.ExpireAtDate` and `.ExpireAtTime`?
- [MinYear](https://learn.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.datepicker.minyear?view=windows-app-sdk-1.2#microsoft-ui-xaml-controls-datepicker-minyear) and [MaxYear](https://learn.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.datepicker.maxyear?view=windows-app-sdk-1.2#microsoft-ui-xaml-controls-datepicker-maxyear)
- maybe `YearVisible="false"`, but I wonder how that will work/look at the end of the year?
- should we show a preference for 12-hour or 24-hour clocks?

Reply:
Not sure about the `YearVisible` setting here. Feels like it should be here just because we're operating with full dates and time. If I want to set the awake state for the next 2 years, why shouldn't I? 😊

I've also cleaned up the naming a bit to make it consistent.

Re: hours, 24 hours is likely best, good call-out.